frontend/catalyst/jax_tracer.py

```diff
@@ -151,7 +151,10 @@ class Function:
     @debug_logger_init
     def __init__(self, fn):
         self.fn = fn
-        self.__name__ = fn.__name__
+        if isinstance(fn, partial):
+            self.__name__ = fn.func.__name__
```

Comment:
This is a bit too specific for my taste, although it works. We might want to consider what to do with other callables that don't define a `__name__` attribute :)

Reply:
Maybe always using ``repr`` to get a printable string of the given function?

Reply:
During QA this week, I've run into a couple of situation where I've gotten the error

```python
AttributeError: '?' object has no attribute '__name__'
```

I can't remember exactly what the ? object was though, or how I managed to trigger it :)